plugins/logitech-tap/fu-logitech-tap.rs

```diff
@@ -34,3 +34,68 @@ struct FuStructLogitechTapTouchHidReq {
     cmd: FuStructLogitechTapTouchHidCmd,
     //payload goes here
 }
+
+#[repr(u8)]
+enum FuLogitechTapSensorHidTdeMode {
+    Disable = 0x0,
+    Enable = 0x01,
+    Selector = 0x02,
+}
+
+// device version
+#[repr(u8)]
+enum FuLogitechTapSensorHidColossusApp {
+    GetVersion = 0x04,
+}
+
+// serial number of the device
+#[repr(u8)]
+enum FuLogitechTapSensorHidSerialNumber {
+    SetReportByte1 = 0x0,
+    SetReportByte4 = 0x0,
+    SetReportByte3 = 0x0E,
+    SetReportByte2 = 0x70,
+}
+
+#[repr(u8)]
+enum FuLogitechTapSensorHidReboot {
+    PinClr = 0x05,
+    PinSet = 0x06,
+    Pwr = 0x2D,
+    Rst = 0x2E,
+}
+
+ #[repr(u8)]
+ enum FuLogitechTapSensorHidSetCmd {
+     // put device into suspend/operational mode
+     Tde = 0x1A,
+     Reboot = 0x1A,
+     Version = 0x1B,
+     SerialNumber = 0x1C,
+ }
+
+ #[repr(u8)]
+ enum FuLogitechTapSensorHidGetCmd {
+     Version = 0x19,
+     SerialNumber = 0x1D,
+ }
+
+ #[derive(New)]
+ struct FuStructLogitechTapSensorHidReq {
+     cmd: FuLogitechTapSensorHidSetCmd,
+      //payload goes here
+      payload_byte1: u8,
+      payload_byte2: u8,
+      payload_byte3: u8 = 0x0,
+      payload_byte4: u8 = 0x0,
+ }
+
+ #[derive(New)]
+ struct FuStructLogitechTapSensorHidRes {
+     cmd: FuLogitechTapSensorHidGetCmd,
+      //payload goes here
+      payload_byte1: u8 = 0x0,
+      payload_byte2: u8 = 0x0,
+      payload_byte3: u8 = 0x0,
+      payload_byte4: u8 = 0x0,
```

Comment:
you don't need the = 0x0 here -- it's memset to zero by default. Then squash down and rebase and I think it's good -- thanks!

Reply:
Unfortunately,  we are hitting the same regression issue again (No detected devices) with this video4linux/hidraw device.

Everything works fine when we use 1.9.24 codebase. But same device, on same platform/setup fails, with latest  codebase.

vv logs not helping much. Plugin is getting invoked but not device_init()/device_class_init(). Any pointers!

Reply:
> with latest codebase

Do you mean the `main` branch or `wip/vcdmp/tap_sensor_rust_refactor_v0`? Can you share the -vv logs again please on the failing branch. Thanks.

Reply:
seeing regression with main branch. vv logs attached.
[fwupdtool_vv.zip](https://github.com/user-attachments/files/17282556/fwupdtool_vv.zip)

Reply:
> 17:36:17.854 FuEngine             failed to setup backend udev: failed to set up netlink: bind to udev socket failed: Address already in use

Your build directory is linking in old libraries. Do something like `mv venv venv.old` and then re-do `./contrib/setup` and it should work again.

Reply:
> $ fwupdtool --plugins logitech_tap get-devices --verbose --verbose

You're not using sudo? Does it work if you use that? Did it work before if you didn't use sudo?

Reply:
> You're not using sudo? Does it work if you use that? Did it work before if you didn't use sudo?

`venv` does it for you.

Reply:
@superm1 for the `failed to set up netlink: bind to udev socket failed: Address already in use` is there any in-process tool to report *what* is binding to the same socket? I've not seen this in a loong time, and there shouldn't be anything linking against gudev at all in fwupd now.

Reply:
Well so I saw the exact same thing on one of my machines... IIRC rebooted it and it _seemed_ to recover.  But knowing that it's happening to another person I'm not so sure what could be going on.

Reply:
I wonder if venv is affecting the linking -- i.e. does `sudo ./src/fwupdtool get-devices` also do the same?

Reply:
@vcdmp if you do `sudo gdb --args ./src/fwupdtool --plugins logitech_tap get-devices --verbose --verbose` and then `b device_monitor_new_full` then `r` -- do you get a breakpoint stop? If so, what does `bt` say when the breakpoint has been hit?

Reply:
Yeah I'd say axe the whole `venv` directory to rule it out.  `./contrib/setup` will rebuild it.

Reply:
Did I missed step here? I tried both, within venv and without venv"
ubuntu22@ubuntu22-ThinkPad-P1:~/repo/fwupd$ v4l2-ctl --list-devices
Synaptics RMI4 Touch Sensor (rmi4:rmi4-00.fn54):
	/dev/v4l-touch0

Logitech Tap HDMI Capture (usb-0000:00:14.0-1.2):
	/dev/video4
	/dev/video5
	/dev/media2

Integrated IR Camera: Integrate (usb-0000:00:14.0-12):
	/dev/video2
	/dev/video3
	/dev/media1

Integrated Camera: Integrated C (usb-0000:00:14.0-8):
	/dev/video0
	/dev/video1
	/dev/media0

ubuntu22@ubuntu22-ThinkPad-P1:~/repo/fwupd$ sudo gdb --args ./src/fwupdtool --plugins logitech_tap get-devices --verbose --verbose
[sudo] password for ubuntu22: 
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04.2) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
./src/fwupdtool: No such file or directory.
(gdb) b device_monitor_new_full
No symbol table is loaded.  Use the "file" command.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (device_monitor_new_full) pending.
(gdb) r --
Starting program:  --
No executable file specified.
Use the "file" or "exec-file" command.
(gdb)

Reply:
Usually I do "DEBUG=1 fwupdtoo...." for live debugging

Reply:
FYI, observing same issue on last release branch "2.0.0".

Reply:
>  ./src/fwupdtool: No such file or directory.

I think you need to do `cd venv/build` to run that directly. 

> Usually I do "DEBUG=1 fwupdtoo...." for live debugging

If that runs gdb, that's fine too. I need the `bt` to see what's linking in udev. Thanks.

Reply:
In the meantime, to unblock your testing, can you try the fix in https://github.com/fwupd/fwupd/pull/7869 i.e. just cherry pick 94edef0b74177fcf16743608e80ceb8f30404e00 please.

Reply:
@hughsie Recloned main branch to pickup your recent [changes](https://github.com/fwupd/fwupd/pull/7869).  

Seems like it is working when invoking fwupdtool directly from build folder.
Same setup fails with "No detected devices" error, when fwupdtool invoked inside virtual environment. 
I have attached vv logs for both usecases.

[fwupdtool_vv.zip](https://github.com/user-attachments/files/17297014/fwupdtool_vv.zip)